packages/clerk-js/src/ui/components/UserProfile/EnterpriseAccountsSection.tsx

```diff
@@ -0,0 +1,78 @@
+import type { SamlAccountResource } from '@clerk/types/src';
+
+import { useRouter } from '../../../ui/router';
+import { useCoreUser } from '../../contexts';
+import { Badge, Col, descriptors, Flex, Image, localizationKeys } from '../../customizables';
+import { ProfileSection, UserPreview } from '../../elements';
+import { useSaml } from '../../hooks';
+import { UserProfileAccordion } from './UserProfileAccordion';
+
+export const EnterpriseAccountsSection = () => {
+  const user = useCoreUser();
+
+  return (
+    <ProfileSection
+      title={localizationKeys('userProfile.start.enterpriseAccountsSection.title')}
+      id='enterpriseAccounts'
+    >
+      {user.samlAccounts.map(account => (
+        <EnterpriseAccountAccordion
+          key={account.id}
+          account={account}
+        />
+      ))}
+    </ProfileSection>
+  );
+};
+
+const EnterpriseAccountAccordion = ({ account }: { account: SamlAccountResource }) => {
+  const router = useRouter();
+  const { getSamlProviderLogoUrl, getSamlProviderName } = useSaml();
+  const error = account.verification?.error?.longMessage;
+  const label = account.emailAddress;
+  const providerName = getSamlProviderName(account.provider);
+  const providerLogoUrl = getSamlProviderLogoUrl(account.provider);
+
+  return (
+    <UserProfileAccordion
+      onCloseCallback={router.urlStateParam?.clearUrlStateParam}
+      icon={
+        <Image
+          elementDescriptor={[descriptors.providerIcon]}
+          elementId={descriptors.enterpriseButtonsProviderIcon.setId(account.id)}
```

Comment:
What is the purpose of this `elementId`? This will create an extra class with the account id as a suffix. Is that what we want?

Reply:
Resolved as discussed in our huddle.